### PR TITLE
Schedule constraints UI improvements

### LINKED ISSUE #41: Move schedule constraints to the top of the planning page and display the configuration up front
- move start date, end date and budget to the top of the page
- add new copy (see image for copy changes) 
- Remove the schedule constraints button 
- Add a confirmation button to let user confirm their variables. 
- [to discuss] Before hitting confirm, grey out the other components of the block? cc @boazsender @susanpark wdyt of this behavior since we're adding a confirmation button? 

image

### LINKED ISSUE #43: Move schedule constraints to the top of the planning page and display the configuration up front
- move start date, end date and budget to the top of the page
- add new copy (see image for copy changes)
- Remove the schedule constraints button
- Add a confirmation button to let user confirm their variables.
- [to discuss] Before hitting confirm, grey out the other components of the block? cc @boazsender @susanpark wdyt of this behavior since we're adding a confirmation button? Alternatively, if it's easier we could update in real-time. 

https://user-images.githubusercontent.com/7271877/82085004-2b628780-96a1-11ea-93ce-0c97a2bb80c6.png

### PR DESCRIPTION
**Pull request by arebe**
Closes #43 

As per design v2: new `Constraints` UI: Move schedule and budget constraints to top of planning page, add new copy, add confirmation button. 

To create the `Confirm` button UX - I refactored the `DndProvider` by moving it to wrap the top-level `Chooser` component to resolve rendering / HTML5 Backend error.

There's a new field in `Constraints` for max distance - while it appears in the UI, it's disabled until the distance calculation functionality build out.

Screen Shot 2020-05-21 at 5 17 36 PM

### CODE REVIEW

File: frontend/constraints.jsx [hunk 1]
```diff
@@ -26,44 +28,96 @@ const trimDate = (date) => date.replace(/T.*$/, '');
...
-            <FormField label="End date" width="50%" paddingLeft={1} style={{float: 'left'}}>
-                <Input
-                    type="date"
-                    value={trimDate(endDate)}
-                    onChange={(event) => onEndDateChange(expandDate(event.target.value))}
-                />
-            </FormField>
+                <Box >
+                    <Text style={{fontWeight: 800}}>Set the dates</Text>
+                    <FormField label="Start date" width="50%" paddingRight={1} style={{float: 'left'}}>
```
Comment (jugglinmike) [inline frontend/constraints.jsx:63]: The spec calls for a label of "Start"

File: frontend/constraints.jsx [hunk 2]
```diff
@@ -26,44 +28,96 @@ const trimDate = (date) => date.replace(/T.*$/, '');
...
-                        style={{width: 'auto'}}
-                        value={String(budget)}
-                        onChange={(event) => {
-                            onBudgetChange(parseFloat(event.target.value));
-                        }}
-                    />
-                </div>
-            </FormField>
-        </Dialog>
+                    <FormField label="End date" width="50%" paddingLeft={1} style={{float: 'left'}}>
```
Comment (jugglinmike) [inline frontend/constraints.jsx:72]: The spec calls for a label of "End"

File: frontend/constraints.jsx [hunk 3]
```diff
@@ -26,44 +28,96 @@ const trimDate = (date) => date.replace(/T.*$/, '');
...
+                            style={{width: 'auto'}}
+                            type="date"
+                            value={trimDate(formEndDate)}
+                            onChange={(event) => setFormEndDate(expandDate(event.target.value))}
+                        />
+                    </FormField>
+                </Box>
+                <Box>
+                    <Text style={{fontWeight: 800}}>Set the budget</Text>
+                    <FormField label="Budget" width="100%">
```
Comment (jugglinmike) [inline frontend/constraints.jsx:83]: The spec calls for a label of "Dollar amount"

File: frontend/constraints.jsx (hunk 1 above)
Comment (arebe) [reply frontend/constraints.jsx:63]: good catch, thanks!

**Comment by susanpark** [comment]
> Fixed positioning perpetually assigns some of the viewport to the menu, but the menu may not be relevant during most of a working session. This seems suboptimal for an application where so much information has to be displayed concurrently.

There are various issues with how it's built today in terms of viewport - we've gotten feedback on this from various chapters and we want to prioritize it, as it creates a bad experience
1. When a hospital schedule slot shows up, that means a user has to select a restaurant to match with the hospital on the right side. Right now, the left side height of the restaurant is determined by the how many hospitals appear on the right side, which is unreasonably short. _All_ of our chapters have a large list of restaurants they need to scroll/sort through to match the hospitals. The goal for chapters is to **find a restaurant**, and we need to optimize for the discoverability of a restaurant. When there is only one or two restaurants, it becomes very difficult for chapters to find restaurants. The viewport should be large enough for them to scroll through the restaurants. 
2. We should never surprise users with inconsistency. Just because there is only one or two hospitals on the right side, doesn't mean that we should change the heights of the viewports for depending on the number of hospitals. This kind of inconsistency causes a negative experience and surprises users when they try to set up a schedule.

**Comment by jugglinmike** [comment]
That all sounds good to me, @susanpark! You may be interested in the changes in gh-22.

My question to Jesar was specifically about the way they intended the "schedule constraints" form to interact with the scroll bar. I'm seeking clarification because the design specification doesn't describe what's intended.

Currently, the patch affixes the form to the top of the Block's viewport such that it is present while the user scrolls through the shifts. An alternate interpretation would be to include the form as scrollable content (this would mean it would no longer be visible as the user moved through the shifts).

Here's a screen recording to demonstrate the two interpretations of the design specification:

2020-05-26-ff-viewport

**Comment by jesarshah** [comment]
Hello! Apologies for the delay in my response - based on how people were using it during the few test rounds, I think giving them as much space on the screen to plan is valuable. So after they set the constraints, they spend time dragging restaurants to hospitals. So I think we should allow the constraints and top of the page content to scroll away with the scrollbar. Thanks!

**Comment by jugglinmike** [comment]
Thanks @jesarshah! Thanks @arebe!

**Comment by jugglinmike** [comment]
> also -- it looks like you all are using `merge --squash` strategy, yeah?

Preferably, unless circumstances call for a merge commit

**Pull request merged**